src/it/scala/com/wavesplatform/it/BalancesSpecification.scala

```diff
@@ -0,0 +1,299 @@
+
+      issuedAssetId <- sender.issue(firstAddress, "name", "description", 100000, 2, reissuable = true, fee = 10 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(issuedAssetId))
+
+      reissuedAssetId <- sender.reissue(firstAddress, issuedAssetId, 100000, reissuable = true, fee = 10 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(reissuedAssetId))
+
+      _ <- assertBalances(firstAddress, 140 waves, 40 waves)
+      _ <- assertAssetBalance(firstAddress, issuedAssetId, 200000)
+    } yield succeed
+
+    Await.result(f, 1 minute)
+  }
+
+  test("assets transfer should lead to a change in the asset balance and a change of the waves balance and the effective balance only on the fee") {
+    val f = for {
+      _ <- assertBalances(firstAddress, 140 waves, 40 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+
+      issuedAssetId <- sender.issue(firstAddress, "name", "description", 100000, 2, reissuable = false, fee = 10 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(issuedAssetId))
+
+      transferId <- sender.transfer(firstAddress, secondAddress, 100000, fee = 10 waves, Some(issuedAssetId)).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(transferId))
+
+      _ <- assertBalances(firstAddress, 120 waves, 20 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+
+      _ <- assertAssetBalance(firstAddress, issuedAssetId, 0)
+      _ <- assertAssetBalance(secondAddress, issuedAssetId, 100000)
+    } yield succeed
+
+    Await.result(f, 1 minute)
+  }
+
+  test("waves transfer should lead to a change in the waves balance and the effective balance") {
+    val f = for {
+      _ <- assertBalances(firstAddress, 120 waves, 20 waves)
+      _ <- assertBalances(secondAddress, 100 waves, 200 waves)
+
+      transferId <- sender.transfer(firstAddress, secondAddress, 1 waves, fee = 1 waves).map(_.id)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(transferId))
+
+      _ <- assertBalances(firstAddress, 118 waves, 18 waves)
+      _ <- assertBalances(secondAddress, 101 waves, 201 waves)
+    } yield succeed
+
+    Await.result(f, 1 minute)
+  }
+
+  test("waves payment should lead to a change in the waves balance and the effective balance") {
+    val f = for {
+      _ <- assertBalances(firstAddress, 118 waves, 18 waves)
+      _ <- assertBalances(secondAddress, 101 waves, 201 waves)
+
+      transferId <- sender.payment(firstAddress, secondAddress, 1 waves, fee = 1 waves)
+
+      _ <- Future.traverse(allNodes)(_.waitForTransaction(transferId))
+
+      _ <- assertBalances(firstAddress, 116 waves, 16 waves)
+      _ <- assertBalances(secondAddress, 102 waves, 202 waves)
+    } yield succeed
+
+    Await.result(f, 1 minute)
+  }
+
+  test("assets burn should lead to a change in the asset balance and to a change of the waves balance and the effective balance only on the fee") {
```

Comment:
"burning assets changes issuer's asset balance; issuer's waves balance is decreased by fee"